SwanProjects/src/ProjectWidget.tsx

```diff
@@ -0,0 +1,205 @@
+  const [release, setRelease] = React.useState(defaultRelease);
+
+  const availablePlatforms = options.stacks_options[stack][release];
+  const defaultPlatform = availablePlatforms.includes(options.platform)
+    ? options.platform
+    : availablePlatforms[0];
+  const [platform, setPlatform] = React.useState(defaultPlatform);
+
+  const [userScript, setUserScript] = React.useState(options.user_script || '');
+
+  const onClickSubmit = () => {
+    props.onSubmit({
+      name: projectName,
+      stack,
+      release,
+      platform,
+      user_script: userScript,
+      stacks_options: options.stacks_options // TODO remove this
+    });
+  };
+
+  const onClickCancel = () => {
+    props.onCancel();
+  };
+
+  const onChangeProjectName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setProjectName(event.target.value);
+  };
+
+  const onChangeStack = (newStack: string) => {
+    setStack(newStack);
+    const newRelease = Object.keys(options.stacks_options[newStack])[0];
+    setRelease(newRelease);
+    const newPlatform = options.stacks_options[newStack][newRelease][0];
+    setPlatform(newPlatform);
+  };
+
+  const onChangeRelease = (event: React.ChangeEvent<{ value: unknown }>) => {
+    const newRelease = event.target.value as string;
+    setRelease(newRelease);
+    const newPlatform = options.stacks_options[stack][newRelease][0];
+    setPlatform(newPlatform);
+  };
+
+  const onChangePlatform = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setPlatform(event.target.value as string);
+  };
+
+  const onChangeUserScript = (
+    event: React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setUserScript(event.target.value as string);
+  };
+
+  return (
+    <div className="sw-Dialog-content">
+      <div className="sw-Dialog-project-name">
+        <swanProjectIcon.react className="sw-Dialog-project-icon" tag="span" />
+        <TextField
+          label="Project Name"
+          variant="outlined"
+          onChange={onChangeProjectName}
+          className="sw-Dialog-project-name-input"
+          size="small"
+          required
+          value={projectName}
+        />
+      </div>
+      <div className="sw-Dialog-select-stack">
+        <Card
+          label="LCG"
+          icon={sftIcon}
+          updateCallback={() => onChangeStack('LCG')}
+          isSelected={stack === 'LCG'}
+        />
+        <Card
+          label="CMSSW"
+          icon={cmsIcon}
+          updateCallback={() => onChangeStack('CMSSW')}
+          isSelected={stack === 'CMSSW'}
+        />
+      </div>
+      <div className="sw-Dialog-stack-options">
+        <div>
+          <div className="sw-Dialog-stack-option-label">
+            <span> Release </span>
+            <span className="sw-Dialog-release-tooltip">
+              <HelpTooltip
+                id="release"
+                message="Software stack: TODO! <br/> \
+                                  No message yet.<br/> \
+                                  ;("
```

Comment:
Can this be removed?

Reply:
no, we need to put something meaningful here for the user,
any suggestion?
![tooltip](https://user-images.githubusercontent.com/275148/133458568-5bcc13f9-bf78-4028-a2ef-5eb958712051.png)

Reply:
"The software bundle attached to the project. More info at lcginfo.cern.ch" ?

Reply:
I'm thinking of names: software source? software stack? We currently call it software stack in the start session web form, people should be already familiar with that.

Reply:
"Software stack to use in this project."

Reply:
What about, "Version of the software stack to use in this project", because the software stack itself is selected in the icon,
in this field, the user is selecting the version of the release.
what do you think?

Reply:
I like Enric's one better, version is implicit

Reply:
ok